libraries/utils/src/render-lines-on-sp.ts

```diff
@@ -63,58 +70,66 @@ export class ScatterPlotLinesRenderer {
-                    line.moveTo(aX!, aY!);
-                    const color = this.lines.colors?.[i] ? this.lines.colors?.[i] : '0,128,0';
-                    const opacity = this.lines.opacities?.[i] ? this.lines.opacities?.[i] : 1;
-                    this.ctx.strokeStyle = `rgba(${color},${opacity})`;
-                    this.ctx.lineWidth = this.lines.widths?.[i] ? i === this.currentLineIdx ? this.lines.widths?.[i] + 2 :
+            let pointFromIdx = this.lines.from[i];
+            let pointToIdx = this.lines.to[i];
+            let changedPoints = false;
+            if (pointFromIdx > pointToIdx) {
+                pointFromIdx = this.lines.to[i];
+                pointToIdx = this.lines.from[i];
+                changedPoints = true;
+            }
+            const pointFrom = this.sp.worldToScreen(this.xAxisCol.get(pointFromIdx), this.yAxisCol.get(pointFromIdx));
+            const aX = pointFrom?.x;
+            const aY = pointFrom?.y;
+            const pointTo = this.sp.worldToScreen(this.xAxisCol.get(pointToIdx), this.yAxisCol.get(pointToIdx));
+            const bX = pointTo?.x;
+            const bY = pointTo?.y;
+            if (aX && aY && bX && bY) {
+                const line = new Path2D();
+                this.paths[i] = line;
+                line.moveTo(aX!, aY!);
+                const color = this.lines.color ?? this.lines.colors?.[i] ? this.lines.colors?.[i] : '0,128,0';
+                const opacity = this.lines.opacity ?? this.lines.opacities?.[i] ? this.lines.opacities?.[i] : 1;
+                this.ctx.strokeStyle = `rgba(${color},${opacity})`;
+                this.ctx.lineWidth = this.lines.width ? this.lines.width : this.lines.widths?.[i] ? i === this.currentLineIdx ? this.lines.widths?.[i] + 2 :
                     this.lines.widths?.[i] : i === this.currentLineIdx ? 3 : 1;
-                    let midPoint: DG.Point = this.midPoint(pointFrom, pointTo);
-                    if (this.numberOfLinesPerPair[pairStr] > 1 || linesPerPairCounter[pairStr] > 1) {
-                        midPoint = this.findPerpendicularPointOnCurve(linesPerPairCounter[pairStr],
-                            pointFrom.x, pointFrom.y, midPoint.x, midPoint.y)
-                    }
-                    line.quadraticCurveTo(midPoint.x, midPoint.y, bX, bY);
-                    if (this.lines.drawArrows?.getBit(i))
-                        this.canvasArrow(line, bX, bY, midPoint.x, midPoint.y);
-                    this.ctx.beginPath();
-                    this.ctx.stroke(line);
-                    this.ctx.closePath();
+                let midPoint: DG.Point = this.midPoint(pointFrom, pointTo);
+                if (this.multipleLinesIndices[i]) {
+                    midPoint = this.findPerpendicularPointOnCurve(this.multipleLinesIndices[i],
+                        pointFrom.x, pointFrom.y, midPoint.x, midPoint.y)
                 }
+                line.quadraticCurveTo(midPoint.x, midPoint.y, bX, bY);
```

Comment:
Are we always drawing a quadratic curve, even if it's a straight line? I'm suspecting this is a lot more computationally intensive than just drawing a straight line

Reply:
Changed - in case of straight lines now using ctx.lineTo()